pydantic/types.py

```diff
@@ -1333,12 +1334,45 @@ def encode_str(self, value: str) -> str:
 
 
 @_internal_dataclass.slots_dataclass
-class TransformSchema:
-    """An annotation that can be used to apply a transform to a core schema."""
+class GetPydanticSchema:
+    """A convenience class for creating an annotation that provides pydantic custom type hooks.
 
-    transform: Callable[[CoreSchema], CoreSchema]
+    This class lets you replace the use of a custom type defining one or more of `__get_pydantic_core_schema__`,
+    `__get_pydantic_json_schema__`, and `__prepare_pydantic_annotations__` with a single annotation.
+    """
 
-    def __get_pydantic_core_schema__(
-        self, source_type: type[Any], handler: _annotated_handlers.GetCoreSchemaHandler
-    ) -> CoreSchema:
-        return self.transform(handler(source_type))
+    get_pydantic_core_schema: Callable[[Any, _annotated_handlers.GetCoreSchemaHandler], CoreSchema] | None = None
+    get_pydantic_json_schema: Callable[[Any, _annotated_handlers.GetJsonSchemaHandler], JsonSchemaValue] | None = None
+    prepare_pydantic_annotations: Callable[[Any, tuple[Any, ...], ConfigDict], tuple[Any, Iterable[Any]]] | None = None
```

Comment:
Can we leave this one off for now?

Reply:
I'm okay with that, but I would suggest we leave this line there with a comment explaining that if we find that it would be useful, it could be uncommented.

Reply:
Sure fine by me